andela/haven-ah-backend
# #161342836 Implement email confirmation on user registeration

#### What does this PR do?
Send email for account confirmation after signup.

#### Description of task to be completed
- use nodemailer package for email sending
- add api endpoint to confirm email.
- update http response utility to work in absence of output data.

#### How should this be manually tested?
- on the terminal run  `git clone https://github.com/andela/haven-ah-backend.git`.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on postman
- run `npm run start-dev`.
on postman navigate to localhost:5000/api/users/signup.

*Sample Request Body*
```gherkin
{
  "username": "string",
  "firstName": "string",
  "lastName": "string",
  "email": "string",
  "password": "string",
}
```
On successful signup, an email would be sent with a url `http://localhost:5000/api/v1/auth/confirm/token`. Clicking on the url takes user to the url and sends a request to confirm user email.

*Response Body sample*
```gherkhin
{
  "status": 200,
  "message": "Your email has been confirmed"
}
```
If the account has been previously confirmed, an error **400 Bad Request** is sent.
```gherkhin
{
  "status": 400,
  "message": "This account has been confirmed already."
}
```

#### Any background context you want to provide?
The signup route was previously implemented without email confirmation.

#### What are the relevant pivotal tracker stories?
#161342836

#### Screenshots (if appropriate)
<img width="1073" alt="screen shot 2018-11-01 at 1 33 57 pm" src="https://user-images.githubusercontent.com/29674347/47852329-b93a9280-dddb-11e8-997f-8837618940a1.png">


<img width="1069" alt="screen shot 2018-10-30 at 10 04 19 pm" src="https://user-images.githubusercontent.com/29674347/47750727-4d97de80-dc90-11e8-93ce-c78f6dd75af9.png">

<img width="1066" alt="screen shot 2018-10-31 at 9 54 24 am" src="https://user-images.githubusercontent.com/29674347/47778782-db5ae480-dcf7-11e8-9ce1-7919e69a008e.png">
